### PR TITLE
Fixing bug in MultTranspose in BlockLowerTriangular preconditioner [blockprec_bugfix]

### PR DESCRIPTION
**Pull request by ScottMacLachlan**
There was a mistake in the range of this loop, which this fixes.

### CODE REVIEW

File: linalg/blockoperator.cpp [hunk 1]
```diff
@@ -288,7 +288,7 @@ void BlockLowerTriangularPreconditioner::MultTranspose (const Vector & x,
       tmp2.SetSize(offsets[iRow+1] - offsets[iRow]);
       tmp2 = 0.0;
       tmp2 += xblock.GetBlock(iRow);
-      for (int jCol=iRow+1; jCol < iRow; ++jCol)
+      for (int jCol=iRow+1; jCol < nBlocks; ++jCol)
       {
          if (op(iRow,jCol))
```
Comment (v-dobrev) [inline linalg/blockoperator.cpp]: At this point, `iRow < jCol`, i.e. the upper triangular part of the block matrix is accessed. It looks like `iRow` and `jCol` are block-indices in the transpose matrix, so they just need to be transposed when accessing `op`.

Comment (tzanio) [inline linalg/blockoperator.cpp]: I know this is probably a square operator, but shouldn't these checks be transposed, e.g. `MFEM_ASSERT(x.Size() == height`?

Comment (tzanio) [inline linalg/blockoperator.cpp]: True.

I think I fixed it in the last commit (as well as a few other minor things).

@ScottMacLachlan, can you take a look?

Comment (ScottMacLachlan) [inline linalg/blockoperator.cpp]: Yes - this looks good to me now.  Thanks!

Comment (ScottMacLachlan) [inline linalg/blockoperator.cpp]: Yes - they should probably be transposed, but the setup routine is fixed so that this must be a square operator.

**Pull request merged**